frontend/src/metabase/entities/fields.js

```diff
@@ -88,11 +88,18 @@ const Fields = createEntity({
     }),
 
     updateField(field, opts) {
-      return Fields.actions.update(
-        { id: field.id },
-        field,
-        notify(opts, field.display_name, t`updated`),
-      );
+      return (dispatch, getState) =>
+        Fields.actions
+          .update(
+            { id: field.id },
+            field,
+            notify(opts, field.display_name, t`updated`),
+          )(dispatch, getState)
+          .then(result => {
+            // Field values needs to be fetched again once the field is updated metabase#16322
+            dispatch(Fields.actions.fetchFieldValues(field));
+            return result;
+          });
```

Comment:
Curious if we can make this block a bit more readable? Maybe convert it into an async function and use `await`. Something like:

```js
async updateField(field, opts) {
  const result = await Fields.actions.update({ ... })(dispatch)(getState)
  await dispatch(Fields.actions.fetchFieldValues(field));
  return result;
}
```

Reply:
Yeah, I did that first, then changed it back to just a promise. I'm more fond of promise than `async...await` but in this case I have to agree `async...await` reads a lot better.